### ISSUE TITLE
Cyclic dependencies between alcotest/tyxml/odoc

### ISSUE THREAD

**Issue by Khady** [ocaml/opam-repository#16663]
recently getting this in the CI of a project. I suspect that there is a problem since a recent release (might be odoc 1.5.1?)

```
Run opam depext notabot --yes --with-doc --with-test
# Detecting depexts using vars: arch=x86_64, os=linux, os-distribution=ubuntu, os-family=debian
[ERROR] No solution for notabot: The actions to process have cyclic dependencies:
          - ∗ alcotest.1.1.0 → ∗ tyxml.4.4.0 → ∗ odoc.1.5.1 → ∗ alcotest.1.1.0
```

The list of deps in my project:
```
 "ocaml" {>= "4.03.0"}
 "dune" {>= "2.5.0"}
 "atdgen"
 "base"
 "cmdliner"
 "devkit"
 "hex"
 "lwt"
 "lwt_ppx"
 "nocrypto"
 "stdio"
 "uri"
 "ocamlformat" {dev & = "0.13.0"}
```

**Comment by kit-ty-kate** [comment]
I believe this is fixed in opam 2.1. I reported the same kind of issue here: https://github.com/ocaml-opam/opam-depext/issues/121

The workaround here would be to call opam depext without argument, then with just `--with-test`, then with just `--with-doc`, all separately. In your case however the `--with-test` and `--with-doc` arguments are not necessary since you don't have any doc or test dependencies

**Comment by Khady** [comment]
thanks for the tip
